hudi-examples/hudi-examples-k8s/src/main/java/org/apache/hudi/examples/k8s/quickstart/HudiDataStreamWriter.java

```diff
@@ -0,0 +1,214 @@
+
+        // Enable checkpointing
+        configureCheckpointing(env);
+
+        DataStreamSource<RowData> dataStream = env.addSource(new SampleDataSource());
+
+        final String targetS3Path = System.getenv("TARGET_S3_PATH");
+        HoodiePipeline.Builder builder = createHudiPipeline("hudi_table", createHudiOptions(targetS3Path));
+        builder.sink(dataStream, false);
+
+        env.execute("Api_Sink");
+    }
+
+    /**
+     * Configure Flink checkpointing settings.
+     *
+     * @param env The Flink StreamExecutionEnvironment.
+     */
+    private static void configureCheckpointing(StreamExecutionEnvironment env) {
+        env.enableCheckpointing(5000); // Checkpoint every 5 seconds
+        CheckpointConfig checkpointConfig = env.getCheckpointConfig();
+        checkpointConfig.setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE);
+        checkpointConfig.setMinPauseBetweenCheckpoints(10000); // Minimum time between checkpoints
+        checkpointConfig.setCheckpointTimeout(60000); // Checkpoint timeout in milliseconds
+    }
+
+    /**
+     * Create Hudi options for the data sink.
+     *
+     * @param basePath The base path for Hudi data.
+     * @return A Map containing Hudi options.
+     */
+    private static Map<String, String> createHudiOptions(String basePath) {
+        Map<String, String> options = new HashMap<>();
+        options.put(FlinkOptions.PATH.key(), basePath);
+        options.put(HoodieCommonConfig.HOODIE_FS_ATOMIC_CREATION_SUPPORT.key(), "s3a");
+        options.put(FlinkOptions.TABLE_TYPE.key(), HoodieTableType.MERGE_ON_READ.name());
+        options.put(FlinkOptions.PRECOMBINE_FIELD.key(), "ts");
+        options.put(FlinkOptions.RECORD_KEY_FIELD.key(), "uuid");
+        options.put(FlinkOptions.IGNORE_FAILED.key(), "true");
+        return options;
+    }
+
+    /**
+     * Create a HudiPipeline.Builder with the specified target table and options.
+     *
+     * @param targetTable The name of the Hudi table.
+     * @param options     The Hudi options for the data sink.
+     * @return A HudiPipeline.Builder.
+     */
+    private static HoodiePipeline.Builder createHudiPipeline(String targetTable, Map<String, String> options) {
+        return HoodiePipeline.builder(targetTable)
+                .column("ts TIMESTAMP(3)")
+                .column("uuid VARCHAR(40)")
+                .column("rider VARCHAR(20)")
+                .column("driver VARCHAR(20)")
+                .column("fare DOUBLE")
+                .column("city VARCHAR(20)")
+                .pk("uuid")
+                .partition("city")
+                .options(options);
+    }
+
+    /**
+     * Sample data source for generating RowData objects.
+     */
+    static class SampleDataSource implements SourceFunction<RowData> {
+        private volatile boolean isRunning = true;
+        public static BinaryRowData insertRow(Object... fields) {
+
+            DataType ROW_DATA_TYPE = DataTypes.ROW(
```

Comment:
Maybe we can move the constant row type as a member variable or a separate utility class.